### PR TITLE
Prevent unexpected itemRendered events for Gallery with loop (T759293)

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

File: js/ui/collection/ui.collection_widget.base.js [hunk 1]
```diff
@@ -920,7 +920,9 @@ var CollectionWidget = Widget.inherit({
                 itemIndex: index
             });
 
+
```
Comment (DokaRus) [inline js/ui/collection/ui.collection_widget.base.js]:
```suggestion
```

File: js/ui/collection/ui.collection_widget.base.js [hunk 2]
```diff
@@ -920,7 +920,9 @@ var CollectionWidget = Widget.inherit({
...
             that._executeItemRenderAction(index, itemData, getPublicElement($itemFrame));
+
```
Comment (DokaRus) [inline js/ui/collection/ui.collection_widget.base.js]:
```suggestion
```

File: js/ui/gallery.js [hunk 3]
```diff
@@ -453,15 +453,26 @@ var Gallery = CollectionWidget.inherit({
...
         }
 
         for(i = 0; i < duplicateCount; i++) {
-            this._renderItem(0, items[lastItemIndex - i]).addClass(GALLERY_LOOP_ITEM_CLASS);
+            this._cloneItemForDuplicate($items, lastItemIndex - i, $container);
         }
     },
 
+    _cloneItemForDuplicate: function($items, index, $container) {
+        $items[index] && $($items[index]).clone(true).addClass(GALLERY_LOOP_ITEM_CLASS).css("margin", 0).appendTo($container);
```
Comment (DokaRus) [inline js/ui/gallery.js]:
```suggestion
var item = $items[index];
if(item) {
    $(item)
      .clone(true)
      .addClass(GALLERY_LOOP_ITEM_CLASS)
      .css("margin", 0)
      .appendTo($container);
}
```

File: js/ui/gallery.js [hunk 4]
```diff
@@ -453,15 +453,26 @@ var Gallery = CollectionWidget.inherit({
...
+    },
+
+    _getRealItems: function() {
+        return this.$element().find(".dx-gallery-item:not(." + GALLERY_LOOP_ITEM_CLASS + ")");
```
Comment (DokaRus) [inline js/ui/gallery.js]:
```suggestion
  var selector = "." + GALLERY_ITEM_CLASS + ":not(." + GALLERY_LOOP_ITEM_CLASS + ")";
  return this.$element().find(selector);
```

File: js/ui/gallery.js [hunk 5]
```diff
@@ -453,15 +453,33 @@ var Gallery = CollectionWidget.inherit({
...
+            this._cloneItemForDuplicate($items[i], $container);
         }
 
         for(i = 0; i < duplicateCount; i++) {
-            this._renderItem(0, items[lastItemIndex - i]).addClass(GALLERY_LOOP_ITEM_CLASS);
+            this._cloneItemForDuplicate($items[lastItemIndex - i], $container);
+        }
+    },
+
+    _cloneItemForDuplicate: function(item, $container) {
```
Comment (DokaRus) [inline js/ui/gallery.js:468]:
```suggestion
    _cloneItem: function(item, $container) {
```

File: js/ui/gallery.js [hunk 6]
```diff
@@ -453,15 +453,33 @@ var Gallery = CollectionWidget.inherit({
 
         var duplicateCount = Math.min(this._itemsPerPage(), itemsCount);
 
+        var $items = this._getRealItems();
+        var $container = this._itemContainer();
+
         for(i = 0; i < duplicateCount; i++) {
-            this._renderItem(0, items[i]).addClass(GALLERY_LOOP_ITEM_CLASS);
+            this._cloneItemForDuplicate($items[i], $container);
```
Comment (DokaRus) [inline js/ui/gallery.js:460]:
```suggestion
            this._cloneItem($items[i], $container);
```

**Review by DokaRus — APPROVED** [review]

**Review by KsyushaM — APPROVED** [review]

File: js/ui/gallery.js (hunk 5 above)
Comment (alexander-kotov-dx) [reply js/ui/gallery.js:468]: I think the "_cloneItem" name does not show some function specifics like adding of GALLERY_LOOP_ITEM_CLASS

**Pull request merged**